qiskit_experiments/library/calibration/rough_amplitude_cal.py

```diff
@@ -81,6 +81,9 @@ def __init__(
             auto_update=auto_update,
         )
 
+        self._validate_channels(schedule)
```

Comment:
Why only these calibrations have validation?

Reply:
Rough amp and rough DRAG are the only two that scan a parameter of a `Calibrations` managed schedule. The other cals are all fine cals (iterate gates without changing parameters) or rough frequency (does not accept a schedule).

Reply:
That said, the `_validate_parameters()` call is useless here at the moment because the Rabi and Drag classes both do the same parameter number check in their `__init__` methods which will be called first.

So these validations should be done before calling `super().__init__()` if we want to keep them.

The advantage of keeping them is that they provide a more useful error message if something goes wrong with the schedule retrieval than the default characterization class's exception. The argument for removing them is that they are unlikely to come up very often and without them there will still likely be some exception, just a less precise calibration-oriented one. I don't think it is possible for the parameter one to fail with the way the get_schedule call is formatted. The channel validation could fail but it would take an unusual circumstance (previously adding a schedule to `Calibrations` with fixed channel numbers that do not match the qubit numbers; this case would possibly be better checked in `add_schedule`).